### PR TITLE
Amd64 gsharedvt small sizes

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/mini/memory-access.c [hunk 1]
```diff
@@ -262,7 +262,7 @@ mini_emit_memory_copy_bytes (MonoCompile *cfg, MonoInst *dest, MonoInst *src, Mo
 		mini_emit_memory_barrier (cfg, MONO_MEMORY_BARRIER_SEQ);
 	}
 
-	if ((cfg->opt & MONO_OPT_INTRINS) && (size->opcode == OP_ICONST)) {
+	if ((cfg->opt & MONO_OPT_INTRINS) && (size->opcode == OP_ICONST) && size->inst_c0 < 10000) {
```
Comment (xmcclure) [inline mono/mini/memory-access.c]: I think it would be helpful to have a comment explaining how the number 10,000 was chosen.

Comment (vargaz) [inline mono/mini/memory-access.c]: It was chosen to avoid the assert in the function guarded by the if () :)

**Comment by kumpera** [comment]
Yeah, it's a fix for a regression I introduced. I'll take care of documenting the magic number.

**Comment by xmcclure** [comment]
That all makes sense, just if the magic number appears in memory-access.c then after kumpera's patch the line in memory-access.c should have a code comment or constant name so a casual reader can figure out why.

**Pull request merged**